src/themes/default/pages/Product.vue

```diff
@@ -120,12 +128,25 @@ export default {
       }
     },
     addToCompare () {
-      // todo
-      this.$bus.$emit('notification', {
-        type: 'success',
-        message: 'Product has been added to comparison list. However - this feature is not implemented yet :(',
-        action1: { label: 'OK', action: 'close' }
-      })
+      let self = this
+      if (!self.compare.isCompare) {
+        this.$store.dispatch('compare/addItem', self.product).then(res => {
+          self.compare.isCompare = true
+        })
+      } else {
```

Comment:
Please use ui-store for components state and move the logic to core/pages/Product.vue

We shouldn’t have kept any universal code /  logic within the themes as they can be easily overwritten by end users

Reply:
I got confused with  ui-store, and not sure if I'll do it in the right way. may you fix this and I'll see how you did't for later ☺️